lsp/server/aws-lsp-buildspec/src/language-server/buildSpecServer.ts

```diff
@@ -0,0 +1,134 @@
+import {
+    AwsLanguageService,
+    MutuallyExclusiveLanguageService,
+    SchemaProvider,
+    completionItemUtils,
+    textDocumentUtils,
+} from '@lsp-placeholder/aws-lsp-core'
+import { JsonLanguageService } from '@lsp-placeholder/aws-lsp-json-common'
+import { YamlLanguageService } from '@lsp-placeholder/aws-lsp-yaml-common'
+import {
+    Connection,
+    InitializeParams,
+    InitializeResult,
+    TextDocumentSyncKind,
+    TextDocuments,
+} from 'vscode-languageserver'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+
+export type BuildspecServerProps = {
+    connection: Connection
+    defaultSchemaUri: string
+    schemaProvider: SchemaProvider
+}
+
+/**
+ * This is a demonstration language server that handles both JSON and YAML files according to the
+ * CodeBuild BuildSpec JSON Schema.
+ *
+ * This illustrates how we can wrap LSP Connection calls around a composition of different language services.
+ * In this case, the composition consists of a JSON processor and a YAML processor.
+ */
+export class BuildspecServer {
+    public static readonly serverId = 'aws-lsp-buildspec'
+
+    protected documents = new TextDocuments(TextDocument)
+
+    protected buildSpecService: AwsLanguageService
+
+    protected connection: Connection
+
+    constructor(private readonly props: BuildspecServerProps) {
+        this.connection = props.connection
+
+        this.buildSpecService = new MutuallyExclusiveLanguageService([
```

Comment:
We should be creating the Buildspec Service outside of the server as its own class. We have to keep in mind it is likely the service will be used outside the context of a Language Server (the same way we use the yaml language service ourselves).

Just a `BuildspecService` class which extends MutuallyExclusiveLanguageService, so users don't have to think of how to build an equivalent Buildspec service instance

Reply:
I agree with this - it fits our need to mix and match services so that we aren't fix to a one server to one service approach. I'm planning on making a PR that adds a CloudFormation server in the style you mentioned, for the purpose of comparing these design concepts.